src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/CastParser.java

```diff
@@ -63,8 +63,8 @@ public String parse(boolean isReturn) throws SqlParseException {
             } else if (DataType.valueOf(dataType) == DataType.STRING) {
                 result.add(String.format("def %s = %s.toString()", name, fileName));
             } else if (DataType.valueOf(dataType) == DataType.DATETIME) {
-                result.add(String.format("def %s = new Date(Double.parseDouble(%s.toString()).longValue())",
-                        name, fileName));
+                result.add(String.format("def %s = new SimpleDateFormat(\"yyyy-MM-dd'T'HH:mm:ss.SSS'Z'\")"
+                        + ".parse(%s.toString())", name, fileName));
```

Comment:
So if the format of field value is slightly different from `yyyy-MM-dd'T'HH:mm:ss.SSS'Z'`, ex. `2020-01-10T10:00:00Z`, do we support this case? It would be great if we can support several popular formats because this `CAST` function is not like `STR_TO_DATE` function which can specify format.

Here is what MySQL can support:

```
>create table test (name varchar(255));
>insert into test values ('2020-01-10T10:00:00.000Z');
>insert into test values ('2020-01-10T11:00:00Z');
>insert into test values ('2020-01-11');
>insert into test values ('2020-01-10 12:00:00');
>
>select cast(name as datetime) from test;
cast(name as datetime)
2020-01-10 10:00:00
2020-01-10 11:00:00
2020-01-11 00:00:00
2020-01-10 12:00:00
```

Reply:
Currently we only support `yyyy-MM-dd'T'HH:mm:ss.SSS'Z'` format. I can create an issue to support different formats but I feel that should not be a part of this PR

Reply:
Sounds good! Please also mention `STR_TO_DATE` function in the new issue. Thanks!